src/core/toga/app.py

```diff
@@ -22,6 +23,49 @@
 warnings.filterwarnings("default", category=DeprecationWarning)
 
 
+class Windows(MutableSet):
+    """
+    This class represents windows of a toga app. A window can be added to app
+    by using `app.windows.add(toga.Window(...))` or `app.windows += toga.Window(...)`
+    notations. Adding a window to app automatically sets `window.app` property to the app.
+    """
+    def __init__(self, app, iterable=None):
+        self.app = app
+        self.elements = set() if iterable is None else set(iterable)
+
+    def add(self, window: Window) -> None:
+        if not isinstance(window, Window):
+            raise TypeError("Toga app windows can only add objects of toga.Window type")
+        # Silently not add if duplicate
```

Comment:
What is the best action when we call add/discard for a window that is not associated with the app?

Reply:
Adding a window that isn't associated with the app would be 'normal' behavior. Adding a window if it's already associated with a *different* app should be an exception - same as adding a widget when it's already associated with a different app - but this should be almost impossible to actually do in practice.

I'd argue *removing* a window that isn't associated with this app should be an exception, since it indicates the app has got into a weird state (probably because of something the developer has done tracking their windows).